### PR TITLE
New PR for SUVI 

### PR DESCRIPTION
**Pull request by cbethge**
### Description
See [here](https://github.com/sunpy/sunkit-instruments/pull/52). Opened a new branch because something went horribly wrong with the push, I don't know why.

### CODE REVIEW

**Comment by cbethge** [comment]
Sorry, I had to open a new PR with a new branch.

**Comment by nabobalis** [comment]
> Sorry, I had to open a new PR with a new branch.

Thats fine.

File: sunkit_instruments/suvi/suvi.py [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+
+
+# helper function: parser for the SUVI websites using BeautifulSoup
+def _list_url_directory(url, ext=''):
+    page = requests.get(url).text
+    soup = BeautifulSoup(page, 'html.parser')
+    return [url + node.get('href') for node in soup.find_all('a') if node.get('href').endswith(ext)]
+
+
+def download_data_from_NOAA(date_time, spacecraft=16, wavelength=171, outdir=None,
```
Comment (nabobalis) [inline sunkit_instruments/suvi/suvi.py:121]: I forgot about this but we already do have a client: https://github.com/sunpy/sunpy/blob/main/sunpy/net/dataretriever/sources/goes.py#L174

It should offer the same range of files as this function.

Comment (cbethge) [reply sunkit_instruments/suvi/suvi.py:121]: Thanks, I was unaware of that. I will just remove my function then.

Comment (nabobalis) [reply sunkit_instruments/suvi/suvi.py:121]: You might want to double check, if not, we can update the client in sunpy.

**Comment by nabobalis** [comment]
Sorry about this! I've been swamped at work and not had time to talk to someone at astropy about the fits issue.

**Comment by nabobalis** [comment]
Hi @cbethge, would it be ok if I start going through this PR and pushing changes?

**Comment by cbethge** [comment]
Hi @nabobalis, sorry, just came back from a longer vacation. Sure, go ahead, let me know if there's anything I need to do.

**Comment by nabobalis** [comment]
For now I created a new branch on the repository to not have to force push to this PR until I think it is ready. 

You can see the current state here: https://github.com/sunpy/sunkit-instruments/pull/61

Main thing is that I moved more code around, got rid of some options from functions and created an example out of the plot thematic map function.

Next step are the tests

**Comment by nabobalis** [comment]
I made all the changes in https://github.com/sunpy/sunkit-instruments/pull/61 and I think it will be easier to review over there!

**Comment by nabobalis** [comment]
Closed in favour of https://github.com/sunpy/sunkit-instruments/pull/61

**Pull request closed without merging**